lib/checksum-stream.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const stream = require('stream');
+const crypto = require('crypto');
+const CRC32Stream = require('crc32-stream');
+
+/**
+ * @summary Get a hash stream
+ * @function
+ * @private
+ *
+ * @param {String} algorithm - either `crc32` or anything supported by `crypto.Hash`
+ * @returns {Stream.Transform}
+ */
+const getHashStream = (algorithm) => {
+  if (algorithm === 'crc32') {
+    return new CRC32Stream();
+  }
+  return crypto.createHash(algorithm);
+};
+
+/**
+ * @summary
+ * @name ChecksumStream
+ * @class
+ */
+class ChecksumStream extends stream.Transform {
+
+  /**
+   * @summary Create an instance of ChecksumStream
+   * @name ChecksumStream
+   * @constructor
+   * @param {Object} options - options
+   * @param {String[]} options.algorithms - hash algorithms
+   */
+  constructor(options = {}) {
+    super(options);
+    this.results = {};
+    this.algorithms = options.algorithms || [];
+    this.hashes = this.algorithms.map((algorithm) => {
```

Comment:
I think its fine to be defensive here. Etcher should always pass an algorithm to this module. If there is not one specific for the image we should default to CRC32/MD5.